internal/local/structure_v1.go

```diff
@@ -0,0 +1,237 @@
+		return err
+	}
+	if err := a.unmarshalAuthProviders(rootDir); err != nil {
+		return err
+	}
+	if err := a.unmarshalFunctions(rootDir); err != nil {
+		return err
+	}
+	if err := a.unmarshalTriggers(rootDir); err != nil {
+		return err
+	}
+	if err := a.unmarshalGraphQL(rootDir); err != nil {
+		return err
+	}
+	if err := a.unmarshalServices(rootDir); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (a *AppDataV1) unmarshalSecrets(rootDir string) error {
+	opts := defaultFileOptions(rootDir, FileSecrets)
+	if _, statErr := os.Stat(opts.path); statErr == nil {
+		if err := unmarshalJSONInto(&a.Secrets, opts); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (a *AppDataV1) unmarshalAuthProviders(rootDir string) error {
+	authProviders, err := unmarshalDirectoryFlat(filepath.Join(rootDir, NameAuthProviders))
+	if err != nil {
+		return err
+	}
+	a.AuthProviders = authProviders
+	return nil
+}
+
+func (a *AppDataV1) unmarshalFunctions(rootDir string) error {
+	functions, err := unmarshalFunctionsV1(filepath.Join(rootDir, NameFunctions))
+	if err != nil {
+		return err
+	}
+	a.Functions = functions
+	return nil
+}
+
+func unmarshalFunctionsV1(path string) ([]map[string]interface{}, error) {
+	var out []map[string]interface{}
+
+	dw := directoryWalker{path: path, onlyDirs: true}
+	if walkErr := dw.walk(func(file os.FileInfo, path string) error {
+		if strings.Contains(path, nameNodeModules) {
+			return nil // skip node_modules since we upload that as a single entity
+		}
+
+		var cfg interface{}
+		if err := unmarshalJSONInto(&cfg, defaultFileOptions(path, FileConfig)); err != nil {
+			return err
+		}
+
+		src, srcErr := ioutil.ReadFile(filepath.Join(path, FileSource.String()))
+		if srcErr != nil {
+			return srcErr
+		}
+
+		o := map[string]interface{}{
+			NameConfig: cfg,
+			NameSource: string(src),
+		}
+		out = append(out, o)
+		return nil
+	}); walkErr != nil {
+		return nil, walkErr
+	}
+
+	return out, nil
+}
+
+func (a *AppDataV1) unmarshalGraphQL(rootDir string) error {
+	dir := filepath.Join(rootDir, NameGraphQL)
+
+	if err := unmarshalJSONInto(&a.GraphQL.Config, defaultFileOptions(dir, FileConfig)); err != nil {
+		return err
+	}
+
+	customResolvers := []map[string]interface{}{}
+	dw := directoryWalker{path: filepath.Join(dir, NameCustomResolvers), onlyFiles: true}
+	if walkErr := dw.walk(func(file os.FileInfo, path string) error {
+		var out map[string]interface{}
+		if err := unmarshalJSONInto(&out, fileOptions{path: path}); err != nil {
+			return err
+		}
+
+		customResolvers = append(customResolvers, out)
+		return nil
+	}); walkErr != nil {
+		return walkErr
+	}
+	a.GraphQL.CustomResolvers = customResolvers
+	return nil
+}
+
+func (a *AppDataV1) unmarshalServices(rootDir string) error {
+	dir := filepath.Join(rootDir, NameServices)
+
+	dw := directoryWalker{path: dir, onlyDirs: true}
+	if walkErr := dw.walk(func(file os.FileInfo, path string) error {
+		var out serviceStructure
+
```

Comment:
[mega-nit] extra line

Reply:
fwiw, that was intentional... but i'll tighten it up in here